File: app/src/main/java/com/aivoice/translate/Activity/TextActivity.java
package com.aivoice.translate.Activity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.core.content.ContextCompat;

import com.aivoice.translate.AdsUtils.FirebaseADHandlers.AdUtils;
import com.aivoice.translate.AdsUtils.Interfaces.AppInterfaces;
import com.aivoice.translate.AdsUtils.Utils.Constants;
import com.aivoice.translate.R;
import com.aivoice.translate.base.BaseActivity;
import com.aivoice.translate.translateApi.TranslateApi;
import com.aivoice.translate.utils.Utils;

import butterknife.BindView;
import butterknife.OnClick;

import static com.aivoice.translate.utils.Utils.WHATIS;

public class TextActivity extends BaseActivity {
    @BindView(R.id.mETInput)
    EditText mETInput;
    @BindView(R.id.mIVRecordVoice)
    ImageView mIVRecordVoice;
    @BindView(R.id.mIVText)
    ImageView mIVText;
    @BindView(R.id.mTxtResult)
    TextView mTxtResult;
    @BindView(R.id.mTxtTitle)
    TextView mTxtTitle;
    @BindView(R.id.mTxtFromLang)
    TextView mTxtFromLang;
    @BindView(R.id.mTxtInput)
    TextView mTxtInput;
    @BindView(R.id.mTxtToLang)
    TextView mTxtToLang;
    @BindView(R.id.mIVClear)
    ImageView mIVClear;
    private String mLanguageCodeFrom = "en";
    private String mLanguageCodeTo = "en";
    boolean IsFrom = true;
    public static TextActivity textActivity;

    public static TextActivity getInstance() {
        return textActivity;
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_voice;
    }

    @Override
    protected void initView(Bundle savedInstanceState) {
        AdUtils.showNativeAd(TextActivity.this, Constants.adsJsonPOJO.getParameters().getNative_id().getDefaultValue().getValue(), (LinearLayout) findViewById(R.id.native_ads1), true);

        textActivity = this;
        mIVRecordVoice.setVisibility(View.GONE);
        mIVText.setVisibility(View.VISIBLE);
        mTxtTitle.setText("" + getResources().getText(R.string.text_tran));
        if (!Utils.isEmptyStr(Utils.TEXTEXTRACT)) {
            mTxtResult.setText("");
            mTxtInput.setText("" + Utils.TEXTEXTRACT);
//            mETInput.setSelection(mETInput.getText().length());
            mTxtInput.setVisibility(View.VISIBLE);
            mETInput.setVisibility(View.GONE);
            mIVClear.setVisibility(View.GONE);
        } else {
            mTxtResult.setText(getResources().getString(R.string.textInfo));
            mTxtInput.setVisibility(View.GONE);
            mETInput.setVisibility(View.VISIBLE);
            mIVClear.setVisibility(View.VISIBLE);
        }
//        AdUtils.showNativeAd(StartActivity.this, Constants.adsJsonPOJO.getParameters().getNative_id().getDefaultValue().getValue(), findViewById(R.id.native_ads), true);
    }

    @OnClick({R.id.mIVBack, R.id.mIVTrans, R.id.mRLTranslate, R.id.mTxtFromLang, R.id.mTxtToLang, R.id.mIVClear})
    public void onViewClicked(View view) {
        switch (view.getId()) {
            case R.id.mIVBack:
                AdUtils.showInterstitialAd(TextActivity.this, new AppInterfaces.InterStitialADInterface() {
                    @Override
                    public void adLoadState(boolean isLoaded) {
                        TextActivity.super.onBackPressed();
                    }
                });
                break;
            case R.id.mIVTrans:
                mTransLanguage();
                break;
            case R.id.mRLTranslate:
                mTranslateText();
                break;
            case R.id.mTxtFromLang:
                IsFrom = true;
                WHATIS = "text";
                startActivity(new Intent(textActivity, LanguageActivity.class));
                break;
            case R.id.mTxtToLang:
                IsFrom = false;
                WHATIS = "text";
                startActivity(new Intent(textActivity, LanguageActivity.class));
                break;
            case R.id.mIVClear:
                mETInput.setText("");
                mTxtResult.setText("");
                mETInput.setSelection(0);
                break;

        }
    }

    private void mTransLanguage() {
        String mTmp = mLanguageCodeFrom;
        mLanguageCodeFrom = mLanguageCodeTo;
        mLanguageCodeTo = mTmp;

        String mLanTmp = mTxtFromLang.getText().toString();
        mTxtFromLang.setText(mTxtToLang.getText().toString());
        mTxtToLang.setText(mLanTmp);
    }

    private void mTranslateText() {
        Utils.hideKeyboard(mETInput);
        String input;
        if (!Utils.isEmptyStr(Utils.TEXTEXTRACT)) {
            input = mTxtInput.getText().toString();
        } else {
            input = mETInput.getText().toString();
        }
        Log.e("TAG", "mTranslateText:=====" + mLanguageCodeFrom + "===" + mLanguageCodeTo);
        TranslateApi translate = new TranslateApi();
        translate.setOnTranslationCompleteListener(new TranslateApi.OnTranslationCompleteListener() {
            @Override
            public void onStartTranslation() {
                // here you can perform initial work before translated the text like displaying progress bar
            }

            @Override
            public void onCompleted(String text) {
                // "text" variable will give you the translated text
                mTxtResult.setText(text);
            }

            @Override
            public void onError(Exception e) {

            }
        });
        translate.execute(input, mLanguageCodeFrom, mLanguageCodeTo);
    }

    public void mGetSelctedLang(String lang, String str) {
        if (IsFrom) {
            mLanguageCodeFrom = str;
            mTxtFromLang.setText("" + lang);
        } else {
            mLanguageCodeTo = str;
            mTxtToLang.setText("" + lang);
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        Utils.TEXTEXTRACT = "";
    }

    @Override
    public void onBackPressed() {
        AdUtils.showInterstitialAd(TextActivity.this, new AppInterfaces.InterStitialADInterface() {
            @Override
            public void adLoadState(boolean isLoaded) {
                Utils.TEXTEXTRACT = "";
                finish();
            }
        });

    }
}

File: app/src/main/java/com/texttranslate/voiceimage/TextActivity.kt
package com.texttranslate.voiceimage

import android.os.Bundle
import android.util.Log
import android.view.View
import com.texttranslate.voiceimage.base.PermissionHelperActivity
import com.texttranslate.voiceimage.databinding.ActivityVoiceBinding
import com.texttranslate.voiceimage.interfaces.OnItemClickListener
import com.texttranslate.voiceimage.interfaces.OnTranslationCompleteListener
import com.texttranslate.voiceimage.translateApi.TranslateApi
import com.texttranslate.voiceimage.utils.Utils

class TextActivity : PermissionHelperActivity(), View.OnClickListener, OnItemClickListener {
    private var mLanguageCodeFrom: String? = "en"
    private var mLanguageCodeTo: String? = "en"
    var IsFrom = true
    var textActivity: TextActivity? = null
    private var binding: ActivityVoiceBinding? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityVoiceBinding.inflate(
            layoutInflater
        )
        setContentView(binding!!.getRoot())
        textActivity = this
        binding!!.mIVRecordVoice.setVisibility(View.GONE)
        binding!!.mIVText.setVisibility(View.VISIBLE)
        binding!!.mTxtTitle.text = "" + getResources().getText(R.string.text_tran)
        if (!Utils.isEmptyStr(Utils.TEXTEXTRACT)) {
            binding!!.mTxtResult.text = ""
            binding!!.mTxtInput.text = "" + Utils.TEXTEXTRACT
            //            mETInput.setSelection(mETInput.getText().length());
            binding!!.mTxtInput.visibility = View.VISIBLE
            binding!!.mETInput.visibility = View.GONE
            binding!!.mIVClear.setVisibility(View.GONE)
        } else {
            binding!!.mTxtResult.text = getResources().getString(R.string.textInfo)
            binding!!.mTxtInput.visibility = View.GONE
            binding!!.mETInput.visibility = View.VISIBLE
            binding!!.mIVClear.setVisibility(View.VISIBLE)
        }
        binding!!.mIVBack.setOnClickListener(this)
        binding!!.mIVTrans.setOnClickListener(this)
        binding!!.mRLTranslate.setOnClickListener(this)
        binding!!.mTxtFromLang.setOnClickListener(this)
        binding!!.mTxtToLang.setOnClickListener(this)
        binding!!.mIVClear.setOnClickListener(this)
    }

    var dialog: LanguageDialog? = null
    override fun onClick(v: View) {
        if (v === binding!!.mIVBack) {
            super@TextActivity.finish()
        } else if (v === binding!!.mIVTrans) {
            mTransLanguage()
        } else if (v === binding!!.mRLTranslate) {
            mTranslateText()
        } else if (v === binding!!.mTxtFromLang) {
            IsFrom = true
            Utils.WHATIS = "text"
            dialog = LanguageDialog(this@TextActivity, this)
            dialog!!.show()
        } else if (v === binding!!.mTxtToLang) {
            IsFrom = false
            Utils.WHATIS = "text"
            dialog = LanguageDialog(this@TextActivity, this)
            dialog!!.show()
        } else if (v === binding!!.mIVClear) {
            binding!!.mETInput.setText("")
            binding!!.mTxtResult.text = ""
            binding!!.mETInput.setSelection(0)
        }
    }

    private fun mTransLanguage() {
        val mTmp = mLanguageCodeFrom
        mLanguageCodeFrom = mLanguageCodeTo
        mLanguageCodeTo = mTmp
        val mLanTmp = binding!!.mTxtFromLang.getText().toString()
        binding!!.mTxtFromLang.text = binding!!.mTxtToLang.getText().toString()
        binding!!.mTxtToLang.text = mLanTmp
    }

    private fun mTranslateText() {
        Utils.hideKeyboard(binding!!.mETInput)
        val input: String
        input = if (!Utils.isEmptyStr(Utils.TEXTEXTRACT)) {
            binding!!.mTxtInput.getText().toString()
        } else {
            binding!!.mETInput.getText().toString()
        }
        Log.e("TAG", "mTranslateText:=====$mLanguageCodeFrom===$mLanguageCodeTo")
        val translate = TranslateApi()
        translate.setOnTranslationCompleteListener(object : OnTranslationCompleteListener {
            override fun onStartTranslation() {
                // here you can perform initial work before translated the text like displaying progress bar
            }

            override fun onCompleted(text: String?) {
                // "text" variable will give you the translated text
                binding!!.mTxtResult.text = text
            }

            override fun onError(e: Exception?) {}
        })
        translate.execute(input, mLanguageCodeFrom, mLanguageCodeTo)
    }

    override fun mGetSelctedLang(lang: String?, str: String?) {
        Log.d("TAGAdapter", "mGetSelctedLang: lang " + lang + "str " + str)
        if (IsFrom) {
            mLanguageCodeFrom = str
            binding!!.mTxtFromLang.text = "" + lang
        } else {
            mLanguageCodeTo = str
            binding!!.mTxtToLang.text = "" + lang
        }
        dialog!!.dismiss()
    }

    override fun onDestroy() {
        super.onDestroy()
        Utils.TEXTEXTRACT = ""
    }

    override fun onBackPressed() {
        super.onBackPressed()
        Utils.TEXTEXTRACT = ""
        finish()
    }
}